src/backend/opencl/CMakeLists.txt

```diff
@@ -108,12 +108,26 @@ INCLUDE_DIRECTORIES(${CLFFT_INCLUDE_DIRS})
 LINK_DIRECTORIES(${CLFFT_LIBRARY_DIR})
 
 OPTION(USE_SYSTEM_BOOST_COMPUTE "Use system BoostCompute" OFF)
+
+SET(Compute_INCLUDE_DIRS "")
 IF(USE_SYSTEM_BOOST_COMPUTE)
-    IF(Boost_VERSION VERSION_LESS "1.61")
+    ADD_DEFINITIONS(-DBOOST_ALL_NO_LIB)
+    SET(Boost_USE_STATIC_LIBS OFF)
+    MESSAGE(STATUS "Looking for Boost version >= 1.61.0 ...")
+    FIND_PACKAGE(Boost 1.61.0)
+    IF(Boost_FOUND)
+        MESSAGE(STATUS "Found Boost ${Boost_VERSION}")
+        SET(Compute_INCLUDE_DIRS ${Boost_INCLUDE_DIRS})
+    ELSE()
+        MESSAGE(STATUS "Boost version with Compute headers not found.")
+        MESSAGE(STATUS "Looking for BoostCompute package directly ...")
         FIND_PACKAGE(BoostCompute REQUIRED)
+        SET(Compute_INCLUDE_DIRS ${BoostCompute_INCLUDE_DIRS})
+        MESSAGE(STATUS "Found BoostCompute.")
```

Comment:
I feel like there are a bit too many status messages to see what is actually happening in the CMake code.